src/base/rss/rss_autodownloadrule.cpp

```diff
@@ -117,13 +122,53 @@ namespace RSS
                     && (lastMatch == other.lastMatch)
                     && (savePath == other.savePath)
                     && (category == other.category)
-                    && (addPaused == other.addPaused);
+                    && (addPaused == other.addPaused)
+                    && (smartFilter == other.smartFilter)
+                    && (previouslyMatchedEpisodes == other.previouslyMatchedEpisodes);
         }
     };
 }
 
 using namespace RSS;
 
+static const QString episodeRegex = "(?:^|[^A-Za-z0-9])((?:S\\d{1,}(?:E\\d{1,})+)|(?:\\d{1,}x\\d{1,})|(?:\\d{4}\\.\\d{2}\\.\\d{2}))(?:[^A-Za-z0-9]|$)";
+
+QString computeEpisodeName(const QString &article)
+{
+    QRegExp f(episodeRegex, Qt::CaseInsensitive);
+    // See if we can extract an season/episode number or date from the title
+    if (f.indexIn(article) == -1) {
+        return QString();
+    }
+
+    // Extract out the episode identifier
+    QString matchedEpisode = f.cap(1);
+
+    // Is it in the S01E01 format?
+    if (matchedEpisode.startsWith('S', Qt::CaseInsensitive)) {
```

Comment:
I think you should rework the episodeRegex to used named capture groups for the season and episode portions of each pattern. Then you can get rid of this entire `if/else if` construct and extract the values directly from the match object. It would be something like:
`(?<season>S\d+)(?<ep>E\d+)|(?<season>\d+)x(?<ep>\d+)|...`

You should also set the `DontCaptureOption` in this case so you can use clarifying parens without them interfering with the captures